packages/server/src/task-group.ts

```diff
@@ -0,0 +1,38 @@
+import { Operation, Resource, Task, spawn } from 'effection';
+import assert from 'assert-ts';
+
+export interface TaskGroup extends Resource<void> {
+  spawn<T>(operation: Operation<T>): Operation<Task<T>>;
+}
+
+export function createTaskGroup(name = 'task group'): Resource<TaskGroup> {
+  let tasks = new Set<Task>();
+
+  return {
+    name,
+    *init(scope) {
+      return {
+        *spawn<T>(operation: Operation<T>) {
+          let task: Task<T> | undefined;
+          yield spawn(function*() {
+            let t: Task<T> = task = yield scope.spawn(operation);
```

Comment:
why is there a local `t` and `task` var?

Reply:
Excellent point! It was mainly there to help with the typing because the type of `task` has to be `Task<T> | undefined` and I can't add it to the set or delete it even if it's undefined. In truth though, it is probably better not to have two storage locations in memory when one will suffice.